parser/helper/helpers.go

```diff
@@ -118,16 +172,20 @@ func (h *Helper) GetActorAddressInfo(add address.Address, key filTypes.TipSetKey
 		return addInfo
 	}
 
-	version := tools.VersionFromHeight(h.network, int64(height))
-	addInfo.ActorCid, err = h.actorCache.GetActorCode(add, key, false)
-	if err != nil {
-		h.logger.Errorf("could not get actor code from address. Err: %s", err)
+	if ok, actorCid, actorName := h.isKeylessAccountActor(add, int64(height)); ok {
+		addInfo.ActorCid = actorCid.String()
+		addInfo.ActorType = actorName
 	} else {
-		c, err := cid.Parse(addInfo.ActorCid)
+		addInfo.ActorCid, err = h.actorCache.GetActorCode(add, key, false)
 		if err != nil {
-			h.logger.Errorf("Could not parse params. Cannot cid.parse actor code: %v", err)
+			h.logger.Errorf("could not get actor code from address. Err: %s", err)
+		} else {
+			c, err := cid.Parse(addInfo.ActorCid)
+			if err != nil {
+				h.logger.Errorf("Could not parse params. Cannot cid.parse actor code: %v", err)
+			}
+			addInfo.ActorType, _ = h.GetActorNameFromCid(c, int64(height))
```

Comment:
Don't you think this logic is a bit repetitive with the one on my other comment?